webportal/src/app/job/job-view/job-view.component.js

```diff
@@ -176,7 +177,12 @@ window.loadJobDetail = loadJobDetail;
 
 $('#content-wrapper').html(jobViewHtml);
 $(document).ready(() => {
-  loadJobs();
+  const query = url.parse(window.location.href, true).query;
+  if (query['jobName']) {
+    loadJobDetail(query['jobName']);
+  } else {
+    loadJobs();
```

Comment:
>loadJobs(); [](start = 4, length = 11)

should we show "job not found?"

Reply:
show the empty page if no job returned? what if error occurs in the back-end, should we split those two cases?

Reply:
good idea. show "job not found" if the backend returns no result, and show whatever error code if backend reports error.

---
In reply to: [158775305](https://github.com/Microsoft/pai/pull/139#discussion_r158775305) [](ancestors = 158775305)

Reply:
it's the current implementation.

Reply:
really? I thought if the query['jobname'] return failure you show all the jobs, instead of jobNotFound?

---
In reply to: [158776110](https://github.com/Microsoft/pai/pull/139#discussion_r158776110) [](ancestors = 158776110)

Reply:
if `query[jobName]` is empty, it will show all jobs. otherwise show the specify job.